cg/meta/upload/gt.py

```diff
@@ -43,93 +47,168 @@ def data(self, analysis: Analysis) -> dict:
-                return genotype_file
-        raise FileNotFoundError(f"No vcf or bcf file found for bundle {hk_version_obj.bundle_id}")
-
-    def get_qcmetrics_file(self, hk_version_obj: Version) -> Path:
-        """Fetch a qc_metrics file and return the path"""
-        hk_qcmetrics = self.hk.files(
-            version=hk_version_obj.id, tags=HkMipAnalysisTag.QC_METRICS
-        ).first()
-        LOG.debug(f"Found qc metrics file {hk_qcmetrics.full_path}")
+    def _get_bcf_file(
+        self,
+        case_id: str,
+    ) -> File:
+        """Return a Housekeeper file object.
+        Raises: FileNotFoundError if nothing is found in the Housekeeper bundle."""
+        LOG.debug("Get Genotype files from Housekeeper")
+        genotype_file: File = self._get_genotype_file(case_id)
+        if self._is_variant_file(genotype_file):
+            LOG.debug(f"Found BCF file {genotype_file.full_path}")
+            return genotype_file
+        raise FileNotFoundError(f"No VCF or BCF file found for the last bundle of {case_id}")
+
+    def _get_qcmetrics_file(self, case_id: str) -> Path:
+        """Return a QC metrics file path.
+        Raises: FileNotFoundError if nothing is found in the Housekeeper bundle."""
+        tags: set[str] = HkAnalysisMetricsTag.QC_METRICS
+        hk_qcmetrics: File = self.hk.get_file_from_latest_version(bundle_name=case_id, tags=tags)
+        if not hk_qcmetrics:
+            raise FileNotFoundError("QC metrics file not found for the given Housekeeper version.")
+        LOG.debug(f"Found QC metrics file {hk_qcmetrics.full_path}")
         return Path(hk_qcmetrics.full_path)
 
     @staticmethod
-    def get_parsed_qc_metrics_data(qc_metrics: Path) -> MIPMetricsDeliverables:
-        """Parse the information from a qc metrics file"""
+    def _get_parsed_qc_metrics_deliverables_mip_dna(
+        qc_metrics_file: Path,
+    ) -> MIPMetricsDeliverables:
+        """Parse and return a QC metrics file."""
         qcmetrics_raw: dict = ReadFile.get_content_from_file(
-            file_format=FileFormat.YAML, file_path=qc_metrics
+            file_format=FileFormat.YAML, file_path=qc_metrics_file
         )
         return MIPMetricsDeliverables(**qcmetrics_raw)
 
-    def upload(self, data: dict, replace: bool = False):
-        """Upload data about genotypes for a family of samples."""
-        self.gt.upload(str(data["bcf"]), data["samples_sex"], force=replace)
+    def _get_samples_sex_raredisease(self, case: Case) -> dict[str, dict[str, str]]:
+        """Return sex information from StatusDB and from analysis prediction."""
+        qc_metrics_file: Path = self._get_qcmetrics_file(case_id=case.internal_id)
+        samples_sex: dict[str, dict[str, str]] = {}
+        for sample in case.samples:
+            sample_id: str = sample.internal_id
+            samples_sex[sample_id] = {
+                "pedigree": sample.sex,
+                "analysis": UploadGenotypesAPI._get_analysis_sex_raredisease(
```

Comment:
```suggestion
                "analysis": self._get_analysis_sex_raredisease(
```
Is there a reason for not calling it via self?

Reply:
No